contracts/modules/staking/StakingProducts.sol

```diff
@@ -518,15 +517,19 @@ contract StakingProducts is IStakingProducts, MasterAwareV2, Multicall {
 
   function _setInitialProducts(uint poolId, ProductInitializationParams[] memory params) internal {
 
-    uint globalMinPriceRatio = cover().getGlobalMinPriceRatio();
+    uint numProducts = params.length;
+    uint[] memory productIds = new uint[](numProducts);
+    for (uint i = 0; i < numProducts; i++) {
+      productIds[i] = params[i].productId;
+    }
+    uint[] memory minPriceRatios = coverProducts().getMinPrices(productIds); 
     uint totalTargetWeight;
 
     for (uint i = 0; i < params.length; i++) {
 
       ProductInitializationParams memory param = params[i];
-
-      if (params[i].targetPrice < globalMinPriceRatio) {
-        revert TargetPriceBelowGlobalMinPriceRatio();
+      if (params[i].targetPrice < minPriceRatios[i]) {
+        revert TargetPriceBelowMinPriceRatio();
```

Comment:
we should just iterate once instead of twice since we are iterating the same array:

```solidity
    uint[] memory productIds = new uint[](numProducts);
    for (uint i = 0; i < numProducts; i++) {
        productIds[i] = params[i].productId;
        
        ProductInitializationParams memory param = params[i];
        uint minPriceRatio = coverProducts().getMinPrices(productIds)[i];
        
        if (param.targetPrice < minPriceRatio) {
            revert TargetPriceBelowMinPriceRatio();
        }

        if (param.targetPrice > TARGET_PRICE_DENOMINATOR) {
            revert TargetPriceTooHigh();
        }

        if (param.weight > WEIGHT_DENOMINATOR) {
            revert TargetWeightTooHigh();
        }
```